gateway/ln-gateway/src/lib.rs

```diff
@@ -257,28 +257,19 @@ impl Gateway {
                                     scid_to_federation.clone(),
                                     tg.clone(),
                                 )
-                                .await;
-
-                                match gateway {
-                                    Ok(gateway) => {
-                                        info!("Successfully created Gateway");
-
-                                        let tx = run_webserver(password.clone(), listen, gateway)
-                                            .await
-                                            .expect("Failed to start webserver");
-                                        info!("Successfully started webserver");
-
-                                        Self::handle_htlc_stream(stream, ln_client, handle.clone(), scid_to_federation.clone(), clients.clone()).await;
-                                        warn!("HTLC Stream Lightning connection broken. Stopping webserver...");
-                                        if let Err(e) = tx.send(()).await {
-                                            error!("Error shutting down gatewayd webserver: {e:?}");
-                                        }
-                                    }
-                                    Err(e) => {
-                                        error!("Failed to create Gateway. Waiting 5 seconds and trying again");
-                                        debug!("Error: {e:?}");
-                                        sleep(Duration::from_secs(5)).await;
-                                    }
+                                .await.expect("Failed to created Gateway");
+
+                                info!("Successfully created Gateway");
+
+                                let tx = run_webserver(password.clone(), listen, gateway)
+                                    .await
+                                    .expect("Failed to start webserver");
+                                info!("Successfully started webserver");
+
+                                Self::handle_htlc_stream(stream, ln_client, handle.clone(), scid_to_federation.clone(), clients.clone()).await;
+                                warn!("HTLC Stream Lightning connection broken. Stopping webserver...");
```

Comment:
This log is a little confusing. We're shutting down the whole gateway at this point, right?

Reply:
We shut down the webserver and drop the `Gateway` struct, but `Gatewayd` doesn't stop. Maybe I should change the message to be a little clearer?

Reply:
I guess this is in a loop so webserver should get restarted again?

Reply:
Once it successfully connects to the lightning node, yes the webserver will come back up.